### ISSUE TITLE
rndc works, lego not, how to solve?

### ISSUE THREAD

**Issue by jvw1954** [go-acme/lego#908]
```console
$ sudo cat /etc/bind/rndc.conf
                 server [cut] {
                   key         rndc-key;
                   addresses   { 136.[cut] port 953; };
                 };
                key "rndc-key" {
                  algorithm hmac-md5;
                  secret "[cut]";
                }; 
```
           
```console
$ sudo rndc  -s ns3.[cut] -p 953 -k rndc-key -V status
create memory context
create socket manager
create task manager
create task
create logging context
setting log tag
creating log channel
enabling log channel
create parser
get key for server
get config key list
decode base64 secret
status
post event
using server [cut] ([cut]#953)
create socket
bind socket
connect
create message
render message
schedule recv
send message
parse message
create message
render message
schedule recv
send message
parse message
version: BIND 9.10.3-P4-Debian <id:ebd72b3> (dns)
boot time: Wed, 19 Jun 2019 13:49:14 GMT
last configured: Wed, 19 Jun 2019 13:49:14 GMT
CPUs found: 1
worker threads: 1
UDP listeners per interface: 1
number of zones: 109
debug level: 0
xfers running: 0
xfers deferred: 0
soa queries in progress: 0
query logging is OFF
recursive clients: 0/0/1000
tcp clients: 5/100
server is up and running

```console
$ sudo cat /var/lib/acme/lego.sh
/usr/bin/docker -D run --env-file=/var/lib/acme/environment.sh \
  -ti --rm --name lego -v /var/lib/acme/.lego/:/var/lib/acme/.lego/ \
  goacme/lego --domains [cut] --accept-tos --email root@[cut] --path /var/lib/acme/.lego/ --dns rfc2136 run
```
```console
$ sudo cat /var/lib/acme/environment.sh
#RFC2136_NAMESERVER="136.[cut]:953"
RFC2136_NAMESERVER="[cut]:953"
RFC2136_TSIG_ALGORITHM="hmac-md5.sig-alg.reg.int."
RFC2136_TSIG_KEY="rndc-key"
RFC2136_TSIG_SECRET="[cut]"
```

```console
$ sudo -u acme /var/lib/acme/lego.sh
2019/06/19 14:16:21 [INFO] [cut] acme: Obtaining bundled SAN certificate
2019/06/19 14:16:22 [INFO] [cut] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/[cut]
2019/06/19 14:16:22 [INFO] [cut] acme: Could not find solver for: tls-alpn-01
2019/06/19 14:16:22 [INFO] [cut] acme: Could not find solver for: http-01
2019/06/19 14:16:22 [INFO] [cut] acme: use dns-01 solver
2019/06/19 14:16:22 [INFO] [cut] acme: Preparing to solve DNS-01
2019/06/19 14:16:22 [INFO] [cut] acme: Cleaning DNS-01 challenge
2019/06/19 14:16:22 [WARN] [cut] acme: error cleaning up: rfc2136: failed to remove: could not find the start of authority for _acme-challenge.[cut].: dial udp: address udp/953": unknown port 
2019/06/19 14:16:22 Could not obtain certificates:
        acme: Error -> One or more domains had a problem:
[cut] [cut] acme: error presenting token: rfc2136: failed to insert: could not find the start of authority for _acme-challenge.[cut].: dial udp: address udp/953": unknown port
DEBU[0004] [hijack] End of stdout       
```

**Comment by ldez** [comment]
```
 --dns.resolvers value        Set the resolvers to use for performing recursive DNS queries. Supported: host:port. The default is to use the system resolvers, or Google's DNS resolvers if the system's cannot be determined.
```

**Comment by jvw1954** [comment]
`sudo cat /var/lib/acme/lego.sh`
/usr/bin/docker -D run --env-file=/var/lib/acme/environment.sh \
  -ti --rm --name lego -v /var/lib/acme/.lego/:/var/lib/acme/.lego/ \
  goacme/lego --domains custard.shrl.nl --accept-tos --email root@custard.shrl.nl --path /var/lib/acme/.lego/ --dns rfc2136 \
  --dns.resolvers 10.0.0.159 run

`sudo -u acme /var/lib/acme/lego.sh`
2019/06/19 15:50:45 [INFO] [<cut>] acme: Obtaining bundled SAN certificate
2019/06/19 15:50:46 [INFO] [<cut>] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/<cut>
2019/06/19 15:50:46 [INFO] [<cut>] acme: Could not find solver for: tls-alpn-01
2019/06/19 15:50:46 [INFO] [<cut>] acme: Could not find solver for: http-01
2019/06/19 15:50:46 [INFO] [<cut>] acme: use dns-01 solver
2019/06/19 15:50:46 [INFO] [<cut>] acme: Preparing to solve DNS-01
2019/06/19 15:50:46 [INFO] [<cut>] acme: Cleaning DNS-01 challenge
2019/06/19 15:50:46 [WARN] [<cut>] acme: error cleaning up: rfc2136: failed to remove: could not find the start of authority for _acme-challenge.<cut>.: dial udp: address udp/953": unknown port
2019/06/19 15:50:46 Could not obtain certificates:
        acme: Error -> One or more domains had a problem:
[<cut>] [<cut>] acme: error presenting token: rfc2136: failed to insert: could not find the start of authority for _acme-challenge.<cut>.: dial udp: address udp/953": unknown port
DEBU[0010] [hijack] End of stdout

**Comment by jvw1954** [comment]
adding "allow-recursion { localhost; localnets;};" to /etc/bind/named.conf.options did not solve a thing :-(

**Comment by jvw1954** [comment]
adding "172.0.0.0/16;" (docker ip's) to acl "local_net" did no solve a thing

**Comment by alexandrestein** [comment]
I have a similar issue. 
But I think to know where it comes from.

In our infrastructure we reject UDP packets going to the web.
It seems like the Lego tries to connect directly to the DNS provider directly which is not possible in our case.

Would it be possible to add an option to prevent direct query to the DNS provider and use the system resolver?

Thanks

**Comment by ldez** [comment]
Fixed by #1843
